src/lib389/lib389/backend.py

```diff
@@ -510,6 +510,82 @@ def _lint_mappingtree(self):
             result['items'] = [bename, ]
             yield result
 
+
+    def _lint_backend_implementation(self):
+        backend_impl = self._instance.get_db_lib()
+        if backend_impl == 'bdb':
+            result = DSBLE0006
+            result['items'] = [self.lint_uid()]
+            yield result
+
+
+    def _lint_backend_implementation_mismatch(self):
+        backend_impl = self._instance.get_db_lib()
+        db_files = os.listdir(self._instance.ds_paths.db_dir)
+        if self._instance.ds_paths.db_home_dir is not None and self._instance.ds_paths.db_home_dir != self._instance.ds_paths.db_dir:
+            db_files.extend(os.listdir(self._instance.ds_paths.db_home_dir))
+        db_files = sorted(set(db_files))
+
+        mdb_files = ['data.mdb', 'INFO.mdb', 'lock.mdb']
+        bdb_files = ['__db.001', 'DBVERSION', '__db.003', 'userRoot', 'log.0000000001', '__db.002']
+        mdb_files.sort()
+        bdb_files.sort()
+
+        if (backend_impl == 'bdb' and not all(file in db_files for file in bdb_files)) or \
+           (backend_impl == 'mdb' and not all(file in db_files for file in mdb_files)):
+            result = DSBLE0004
+            result['items'] = [self.lint_uid()]
+            yield result
+
+    def _lint_backend_config_attributes(self):
```

Comment:
I am not sure that we should check the config attribute: I am pretty sure that there are leftover after a migration from bdb to lmdb or the opposite. and the attribute are not invalid, they are ignored.

Reply:
Ok. I am  fine about keeping it and use low severity.